### ISSUE TITLE
test failures reported

### ISSUE THREAD

**Issue by dch** [algernon/riemann-c-client#14]
BTW the version committed to FreeBSD ports tree is slightly different from what I proposed, it's here https://github.com/freebsd/freebsd-ports/tree/master/net-mgmt/riemann-c-client/Makefile 
- FreeBSD 11.0R-p0 amd64
- both poudriere (jail environment) or via `make test`

I'm not clear what the constraints are for the tests to succeed; can I gather any more info? I can organise access to the box if that helps btw.

-- built via `make test`

```
Running suite(s): Riemann C client library tests
87%: Checks: 48, Failures: 6, Errors: 0
tests/check_client.c:47:F:Client:test_riemann_client_connect:0: Assertion 'riemann_client_connect (client, RIEMANN_CLIENT_TCP, "127.0.0.1", 5559) == -ECONNREFUSED' failed: errno==60 (Connection refused), expected==61 (Connection refused)
tests/check_client.c:253:F:Client:test_riemann_client_create:0: Assertion '-errno == -ECONNREFUSED' failed: errno==60 (Connection refused), expected==61 (Connection refused)
tests/check_client.c:346:F:Client:test_riemann_client_send_message:0: Assertion 'riemann_client_send_message (client, message) == -0' failed: errno==56 (No error: 0), expected==0 (No error: 0)
tests/check_client.c:548:F:Client:test_riemann_client_send_message_tls:0: Assertion 'riemann_client_send_message (client, NULL) == -EINVAL' failed: errno==57 (Invalid argument), expected==22 (Invalid argument)
tests/check_client.c:586:F:Client:test_riemann_client_recv_message_tls:0: Assertion '(response = riemann_client_recv_message (client)) != ((void *)0)' failed
tests/check_simple.c:110:F:Simple:test_riemann_simple_communicate:0: Assertion 'response != ((void *)0)' failed
FAIL tests/check_libriemann (exit status: 1)
```

```
# during:
===>  Staging for riemann-c-client-1.9.1
gmake[1]: Entering directory '/usr/ports/net-mgmt/riemann-c-client/work/riemann-c-client-riemann-c-client-1.9.1'
gmake --no-print-directory tests/check_libriemann tests/check_symver
depbase=`echo tests/check_libriemann.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
cc -DHAVE_CONFIG_H -I. -I./lib/riemann    -I./lib -I./lib -O2 -pipe -fno-omit-frame-pointer  -fstack-protector -fno-strict-aliasing -I/usr/local/include -D_TH
heck_libriemann.c &&\
mv -f $depbase.Tpo $depbase.Po
/bin/sh ./libtool  --tag=CC   --mode=link cc -I./lib -I./lib -O2 -pipe -fno-omit-frame-pointer  -fstack-protector -fno-strict-aliasing -I/usr/local/include -D
libtool: link: cc -I./lib -I./lib -O2 -pipe -fno-omit-frame-pointer -fstack-protector -fno-strict-aliasing -I/usr/local/include -D_THREAD_SAFE -pthread -fstac
h -Wl,/usr/ports/net-mgmt/riemann-c-client/work/riemann-c-client-riemann-c-client-1.9.1/lib/.libs -Wl,-rpath -Wl,/usr/local/lib
depbase=`echo tests/check_symver.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
cc -DHAVE_CONFIG_H -I. -I./lib/riemann    -I./lib -I./lib -O2 -pipe -fno-omit-frame-pointer  -fstack-protector -fno-strict-aliasing -I/usr/local/include -D_TH
ver.c &&\
mv -f $depbase.Tpo $depbase.Po
/bin/sh ./libtool  --tag=CC   --mode=link cc -I./lib -I./lib -O2 -pipe -fno-omit-frame-pointer  -fstack-protector -fno-strict-aliasing -I/usr/local/include -D
libtool: link: cc -I./lib -I./lib -O2 -pipe -fno-omit-frame-pointer -fstack-protector -fno-strict-aliasing -I/usr/local/include -D_THREAD_SAFE -pthread -fstac
sr/ports/net-mgmt/riemann-c-client/work/riemann-c-client-riemann-c-client-1.9.1/lib/.libs -Wl,-rpath -Wl,/usr/local/lib
FAIL: tests/check_libriemannheck-TESTS
PASS: tests/check_symver
============================================================================
Testsuite summary for riemann-c-client 1.9.1
============================================================================
# TOTAL: 2
# PASS:  1
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0
============================================================================
See ./test-suite.log
Please report to algernon@madhouse-project.org
============================================================================
gmake[3]: *** [Makefile:1273: test-suite.log] Error 1
gmake[2]: *** [Makefile:1381: check-TESTS] Error 2
gmake[1]: *** [Makefile:1591: check-am] Error 2
gmake[1]: Leaving directory '/usr/ports/net-mgmt/riemann-c-client/work/riemann-c-client-riemann-c-client-1.9.1'
*** Error code 1

Stop.
```

full logs:

https://pkg.skunkwerks.at/poudriere/data/11_amd64-default/2016-10-04_12h55m55s/logs/errors/riemann-c-client-1.9.1.log

**Comment by algernon** [comment]
These all seem to be tests that are guarded by a `network_tests_enabled()`, which  tries to connect to `localhost:5555`, and disables these tests if it can't. So it seems that either there is something running on that port, or the logic fails on FreeBSD for some odd reason.

Can you check if there's something running there?

**Comment by dch** [comment]
Ah yes, this test fails if there's another process listening on 5555 (such as riemann or a forwarder to riemann). Is it possible to choose a random OS provided port for testing? We do this in CouchDB by opening port 0 and then reading the actually assigned port from the socket.

Alternatively if any error returns from the socket check, then skip all network tests?

**Comment by algernon** [comment]
I can make it easily configurable (at `configure`-time) which port(s) to use, if that helps. I'd rather not read from port 0, because there is a config that goes with the test suite, so that should be pre-generated.

Alternatively, I can add a `--disable-network-tests` option to just disable them alltogether. But then, I might just disable them by default, and require a flag to enable them... I think I'll go that route.

Meanwhile, if the library works fine, you can disable running the tests in the ports, for the time being.

**Comment by dch** [comment]
thanks! if you can tag this I'll roll a new release in FreeBSD.

**Comment by algernon** [comment]
FWIW, this is included in the latest 1.10.0 and 1.10.1 releases.
